[CUBRIDMAN-92] Support Synonym (#331)
CUBRID/cubrid-manual: en/sql/schema/synonym_stmt.rst

*****************************
SYNONYM DEFINITION STATEMENTS
*****************************

CREATE SYONYM
=============

Create another new name to use for an object that already exists in the database. The user can use the object with the name specified when creating the database object, and can also use the object as a synonym.

    * The user can only create private synonym, and private synonym can only be used in the schema for the user.

>     * The information of the created synonym can be found in the **db_synonym** system virtual class.

Reviewer: Adding a link to the "db_synonym" could make users help to understand

Author: I agree. I will add it.

Create another new name to use for an object that already exists in the database. The user can use the object with the name specified when creating the database object, and can also use the object as a synonym.

    #. The user can only create private synonym, and private synonym can only be used in the schema for the user.
    #. The information of the created synonym can be found in the :ref:`db_synonym <db_synonym>` system virtual class.
    #. If a table or view already exists with the same name, a synonym cannot be created with that name.

>     #. When creating a synonym, it checks if the schema of the target object exists, but does not check if the target object exists. If the target object does not exist, an error occurs when using a synonym.

Reviewer 2: ```suggestion
    #. When creating a synonym, it checks whether the schema of the target object exists, but it does not check whether the target object exists. If the target object does not exist, an error occurs when using a synonym.
```

Author: Thank you. Changed.

*   *object_name*: Specifies the name of the target object.
*   *synonym_comment_string*: Specifies a comment of a synonym.

.. warning::

    The target object of a synonym supports only tables and views. It is possible to use another object as the target object, but an error occurs when using a synonym.

.. warning::
    
    It does not support public synonym yet.

1. Private synonym
------------------

> In the example below, user u1 and user u2 have created a synonym with the same name, but use different target objects.

Reviewer 2: ```suggestion
In the example below, user u1 and user u2 have created a synonym with the same name but use different target objects.
```

Author: Thank you. Changed.

    ====================================================================================================================================
      's1'                  'U2'                  'NO'                  't2'                  'DBA'                 NULL

3. Synonym name
---------------

If a table or view already exists with the same name, a synonym cannot be created with that name.

.. code-block:: sql

    /* current_user: public */
    create table t1 (c1 varchar(100));
    insert into t1 values ('table for user public.');

    create table s1 (c1 varchar(100));
    create view s2 as select * from t1;

    /* Already used as a table name. */
    create synonym s1 for t1;

.. code-block::

    ERROR: before ' ; '
    Class public.s1 already exists.

.. code-block:: sql

    /* Already used as a view name. */
    create synonym s2 for t1;

.. code-block::

    ERROR: before ' ; '
    Class public.s2 already exists.

.. code-block:: sql

    create synonym s3 for t1;
    select * from s3;

.. code-block::

      c1
    ======================
      'table for user public.'

4. Synonym for synonym
----------------------

> When creating a synonym, it is not checked whether the target object exists, so the user can create a new synonym by specifying an existing synonym as the target object. However, if a synonym is used, the target object of the synonym, which is the target object, is not found again.

Reviewer 2: ```suggestion
When creating a synonym, it does not check whether the target object exists, so the user can create a new synonym by specifying an existing synonym as the target object. However, if a synonym is used, the target object of the synonym, which is the target object, is not found again.
```

Author: Thank you. Changed.
